src/gwt/src/org/rstudio/studio/client/projects/Projects.java

```diff
@@ -605,6 +608,20 @@ public void onOpenProjectEvent(OpenProjectEvent event)
                             event.getAllowOpenInNewWindow());
    }
    
+   @Override
```

Comment:
One thing to confirm here: when creating a new project or switching projects we always prompt for unsaved changes. Just want to make sure that this code path also results in that prompt when appropriate.